File: gams/apifiles/CSharp/Interrupt/Interrupt.cs
using System;
using GAMS;

namespace Interrupt
{
    class Interrupt
    {
        static void Main(string[] args)
        {
            GAMSWorkspace ws = new GAMSWorkspace();

            // Use a MIP that needs some time to solve
            ws.GamsLib("lop");
            GAMSJob job = ws.AddJobFromFile("lop.gms");

            // Change the default behaviour of pressing Ctrl+C to send that signal to a running job
            Console.CancelKeyPress += delegate(object sender, ConsoleCancelEventArgs a) { job.Interrupt(); a.Cancel = true; };

            // Run the job, Pressing Ctrl+C will interrupt the GAMS job, but not this whole application
            job.Run(output: Console.Out);
        }
    }
}

File: gams/apifiles/Python/interrupt.py
from gams import *
import sys
import threading
import time

def interrupt_gams(job):
    time.sleep(2)
    job.interrupt()

if __name__ == "__main__":
    ws = GamsWorkspace()

    # Use a MIP that needs some time to solve
    ws.gamslib("lop")
    job = ws.add_job_from_file("lop.gms")

    # start thread asynchronously that interrupts the GamsJob after 2 seconds
    threading.Thread(target=interrupt_gams, args=(job,)).start()

    # start GamsJob
    job.run(output=sys.stdout)
